tests/regression/symtabAPI/Symtab/file_properties/CMakeLists.txt

```diff
@@ -64,31 +60,29 @@ foreach(strip_t "stripped" "none")
   foreach(pic_t "pic" "none")
     foreach(link_t ${_static_link} "dynlink")
 
-      # static+pic isn't PIE
-      if(link_t STREQUAL "${_static_link}" AND pic_t STREQUAL "pic")
-        continue()
-      endif()
-
       set(_targ "exectest_${strip_t}_${pic_t}_${link_t}")
       make_exec_test(${_targ})
 
-      if(pic_t STREQUAL "pic")
-        set_property(TARGET ${_targ} PROPERTY POSITION_INDEPENDENT_CODE TRUE)
+      if(link_t STREQUAL "${_static_link}")
+        if(pic_t STREQUAL "pic" AND DYNINST_CXX_STATIC_PIE_SUPPORTED)
+          # use static-pie
+          target_link_options(${_targ} PRIVATE ${DYNINST_CXX_STATIC_PIE_FLAG})
+        else()
+          target_link_options(${_targ} PRIVATE ${DYNINST_CXX_STATIC_EXE_LINK_FLAG})
+        endif()
       else()
-        # some compilers use PIE by default
-        target_compile_options(${_targ} PRIVATE "-no-pie")
-        target_link_options(${_targ} PRIVATE "-no-pie")
+        set(_use_pic FALSE)
+        if(pic_t STREQUAL "pic")
+          set(_use_pic TRUE)
+        endif()
+        set_property(TARGET ${_targ} PROPERTY POSITION_INDEPENDENT_CODE ${_use_pic})
```

Comment:
Is there a reason to to just
if  (pic_t STREQUAL "pic")
set_property(...  POSITION_INDEPENDENT_CODE TRUE).
endif()))
and does POSITION_INDEPENDENT_CODE needs to set for static-pie?